src/rust/engine/process_execution/src/bounded.rs

```diff
@@ -269,7 +324,7 @@ impl AsyncSemaphore {
             //
             // This is because we cannot anticipate the number of inbound processes, and we never want to
             // delay a process from starting.
-            let concurrency_desired = max(concurrency_desired, 1);
+            let concurrency_desired = max(max_concurrency, min_concurrency);
```

Comment:
When would this ever not equal `max_concurrency`?

Reply:
It'll always be `max_concurrency` I'll fix this. 

Previously concurrency_desired was allowed to be zero, but now it must be >= 1 so we don't skip acquiring the semaphore. I think it updated this accidentally.